lib/util/propTypes.js

```diff
@@ -547,6 +547,35 @@ module.exports = function propTypesInstructions(context, components, utils) {
     if (node.right) return getRightMostTypeName(node.right);
   }
 
+  /**
+   * Returns true if the node is either a interface or type alias declaration
+   * @param {ASTNode} node
+   * @return {boolean}
+   */
+  function filterInterfaceOrTypeAlias(node) {
+    return (
+      astUtil.isTSInterfaceDeclaration(node) || astUtil.isTSTypeAliasDeclaration(node)
+    );
+  }
+
+  /**
+   * Returns true if the interface or type alias declaration node name matches the type-name str
+   * @param {ASTNode} node
+   * @param {string} typeName
+   * @return {boolean}
+   */
+  function filterInterfaceOrAliasByName(node, typeName) {
+    if (node.id && node.id.name) {
+      return node.id.name === typeName;
+    }
+
+    if (node && node.declaration && node.declaration.id && node.declaration.id.name) {
```

Comment:
if `node` is falsy then this function would have crashed on line 568

Reply:
yes, my bad - the check should not contain `node && ` part. `node` would never be falsy.